controllers/platform/services/services.go

```diff
@@ -138,11 +138,23 @@ func (d DataIndexHandler) MergePodSpec(podSpec corev1.PodSpec) (corev1.PodSpec,
 	return *c, err
 }
 
+// hasPostgreSQLConfigured returns true when either the SonataFlow Platform PostgreSQL CR's structure or the one in the Data Index service specification is not nil
+func (d DataIndexHandler) hasPostgreSQLConfigured() bool {
+	return (d.platform.Spec.Services.DataIndex.Persistence != nil && d.platform.Spec.Services.DataIndex.Persistence.PostgreSql != nil) ||
```

Comment:
No. The purpose of the SonataFlowClusterPlatform is to set cluster-wide configs for **workflow** consumption, NOT for service consumption. This means service configs, e.g. persistence, should only ever be affected by the active platform CR in its own namespace.

To take things a step further, it's important to be *very* careful about what other platform configs are enabled cluster-wide for workflow consumption in the future. For example, In my opinion these new global persistence configs should NOT be enabled cluster-wide for workflows. It would result in a major security risk for users if our operator is providing creds and connectivity information haphazardly to every namespace in the cluster. Persistence should only be set at the namespace level which ensures that only those who should have the creds are using them.

That said, something like `platform.spec.builds` configs might be a good candidate at some point.

Reply:
@tchughesiv then we have to review the API in a follow-up PR. The API **must communicate that**.

For example, everything under ` .spec.local.builds` and `.spec.local.persistence`. Hence, no global config for these attributes. Can you file an issue so we can discuss further after your PR is merged?

Reply:
@ricardozanini i've created an issue to address these concerns. https://github.com/apache/incubator-kie-kogito-serverless-operator/issues/380